### PR TITLE
byteman 3.0.5 (new formula)

### PR DESCRIPTION
**Pull request by msavy**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Contributing a new formula Byteman - http://byteman.jboss.org/

### CODE REVIEW

File: Formula/byteman.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  bottle :unneeded
+  depends_on :java => "1.6+"
+
+  def install
+    env = Language::Java.java_home_env("1.6+").merge(:BYTEMAN_HOME => libexec)
+    rm_rf Dir["bin/*.bat"]
+    doc.install Dir["docs/*"], "README"
+    libexec.install ["bin", "lib", "contrib"]
+    share.install ["sample"]
```
Comment (apjanke) [inline Formula/byteman.rb]: This should probably be `pkgshare` instead of `share`.

File: Formula/byteman.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  desc "Java bytecode manipulation tool for testing, monitoring and tracing."
+  homepage "http://byteman.jboss.org/"
+  url "http://downloads.jboss.org/byteman/3.0.5/byteman-download-3.0.5-bin.zip"
+  sha256 "c7b90b63346d4a0f648c55e12c21110fb173fb9ae8471727fd44906591fa8c26"
+
+  bottle :unneeded
+  depends_on :java => "1.6+"
+
+  def install
+    env = Language::Java.java_home_env("1.6+").merge(:BYTEMAN_HOME => libexec)
```
Comment (apjanke) [inline Formula/byteman.rb]: Maybe move this down to where it's actually used in the wrapper script generation?

**Comment by apjanke** [comment]
Couple minor items on the code; looks good overall. Thanks for including a substantial test in this!

**Comment by msavy** [comment]
@apjanke Done. Thanks for the feedback.

**Comment by apjanke** [comment]
Merged. Thank you for your contribution to Homebrew!

**Pull request closed without merging**